pkgs/development/compilers/sapmachine/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
+
+  pname = "sapmachine";
+
+  version = "13.0.1";
+
+  sha256_linux = "21b2a4d3d80bb8434235e9416d347967ca4714e1c54e49136e739b8447c87e56";
+  sha256_darwin = "f0111c1cba6d1b1042724df7d28f2a274981bd90fbfe6c940ca5cc9ceddd8825";
+
+  platform = if stdenv.isDarwin then "osx" else "linux";
+  sha256 = if stdenv.isDarwin then sha256_darwin else sha256_linux;
+  homepath = if stdenv.isDarwin then "Contents/Home" else ".";
+
+  src = fetchurl {
+    url = "https://github.com/SAP/SapMachine/releases/download/sapmachine-${version}/sapmachine-jdk-${version}_${platform}-x64_bin.tar.gz";
+    sha256 = sha256;
+  };
+
+  buildInputs = [ unzip];
+  
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./$homepath/* "$out/"
+
+    # Set JAVA_HOME
+    mkdir -p $out/nix-support
+    cat <<EOF >> $out/nix-support/setup-hook
+    if [ -z "\$JAVA_HOME" ]; then export JAVA_HOME=$out; fi
+    EOF
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://sap.github.io/SapMachine/;
+    license = licenses.gpl2;
```

Comment:
The gpl license variants without the `Plus` or `Only` suffix are deprecated, because they do not specify the license well enough. Use `Plus` if the project has "or later" sections in the actual source code files (not the license). Otherwise, use the `Only` suffix.